libminifi/src/Connection.cpp

```diff
@@ -169,8 +182,14 @@ void Connection::drain(bool delete_permanently) {
   std::lock_guard<std::mutex> lock(mutex_);
 
   while (!queue_.empty()) {
-    std::shared_ptr<core::FlowFile> item = queue_.pop();
-    logger_->log_debug("Delete flow file UUID %s from connection %s, because it expired", item->getUUIDStr(), name_);
+    // TODO(adebreceni): we don't actually use the whole flow file
+    //  there could be a more optimal solution without triggering a swap-in
```

Comment:
Ideally the swap-in would be done in the content repository, when the flow file is being read. It's enough if we start thinking about smarter solutions when it's apparent that that is too late / too slow.

Reply:
the content in this `drain` method is not accessed, the actual FlowFile properties (e.g. ResourceClaim) are only accessed if `delete_permanently` is `true`, so in theory we should be good with a `FlowFileQueue::clear` method, and only go through (and possibly swap in) FlowFiles when we need to access some properties (for deleting resources from the ContentRepo)

Reply:
added `FlowFileQueue::clear`